doc/rados/troubleshooting/troubleshooting-mon.rst

```diff
@@ -383,6 +383,80 @@ example::
 
 	iptables -A INPUT -m multiport -p tcp -s {ip-address}/{netmask} --dports 6789,6800:7300 -j ACCEPT
 
+Monitor Store Failures
+======================
+
+Symptoms of store corruption
+----------------------------
+
+Ceph monitor stores the `cluster map`_ in a key/value store. If a monitor in a
+cluster fails due to the key/value store corruption, following error messages
+might be found in the monitor log::
+
+  Corruption: error in middle of record
+
+or::
+
+  Corruption: 1 missing files; e.g.: /var/lib/ceph/mon/mon.0/store.db/1234567.ldb
+
+Recovery using healthy monitor(s)
+---------------------------------
+
+If there is any survivers, we can always `replace`_ the corrupted one with a
+new one. And after booting up, the new joiner will sync up with a healthy
+peer, and once it is fully sync'ed, it will be able to serve the clients.
+
+Recovery using OSDs
+-------------------
+
+But what if all monitors fails at the same time? Since users are encouraged to
+deploy at least three monitors in a Ceph cluster, the chance of simultaneous
+failure is rare. But unplanned power-downs in a data centre could fail the
+underlying filesystem, and hence kill all the monitors. In this case, we can
+recover the monitor store with the information stored in OSDs.::
+
+  ms=/tmp/mon-store
+  mkdir $ms
+  # collect the cluster map from OSDs
+  for host in $hosts; do
+    rsync -avz /tmp/mon-store user@host:$ms
```

Comment:
@gregsfortytwo 
> I'd assumed it would be extracting to some intermediate format and then turning them into the mon format at the end.

i also evaluated this approach, it can be executed in parallel when collecting the OSD side info, hence more efficient. but seems current way is simpler. and this fix is only for disaster recovery which is not supposed to be used very frequently. so i am inclined to use this iterative way.